acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/AcceptanceTestBase.java

```diff
@@ -98,6 +108,55 @@ protected AcceptanceTestBase() {
     permissionedNodeBuilder = new PermissionedNodeBuilder();
   }
 
+  static {
+    System.setProperty("log4j2.isThreadContextMapInheritable", "true");
+  }
+
+  @Rule public final TestName name = new TestName();
+
+  @Rule
+  public TestWatcher log_eraser =
+      new TestWatcher() {
+
+        @Override
+        protected void starting(final Description description) {
+          ThreadContext.put("test", description.getMethodName());
+          ThreadContext.put("class", description.getClassName());
+          Thread.currentThread()
+              .setUncaughtExceptionHandler(
+                  (thread, error) ->
+                      LOG.error(
+                          "Uncaught exception in thread \"" + thread.getName() + "\"", error));
+          Thread.setDefaultUncaughtExceptionHandler(
+              (thread, error) ->
+                  LOG.error("Uncaught exception in thread \"" + thread.getName() + "\"", error));
+        }
+
+        @Override
+        protected void failed(final Throwable e, final Description description) {
+          // add the result at the end of the log so it is self-sufficient
+          LOG.error(
+              "==========================================================================================");
+          LOG.error("Test failed. Reported Throwable at the point of failure:", e);
+        }
+
+        @Override
+        protected void succeeded(final Description description) {
+          // if so configured, delete logs of successful tests
+          if (!Boolean.getBoolean("acctests.keepLogs")) {
```

Comment:
`acctests.keepLogs` seems misleading as a variable, as it only applies to successful test executions, not the failed ones?

Reply:
Would `acctests.keepAllLogs` be better?

Reply:
Wouldn't something like `acctests.discardlLogs` make sense, then you could remove the logical inversion?

Reply:
The default behavior as currently coded is to keep logs of failing tests and delete logs of successful tests.
The intended behavior of this variable is to keep all test logs, both failing and successful.
Please suggest a default behavior and which modifier to use :)

Reply:
If failed test logs are kept either way, then is it correct to say that this variable is only about keeping logs for successful / passing tests?

If so, then a name like `accosts.keepLogsOfPassingTests`?

Reply:
Sounds good, done.